extensions/ftp/src/test/java/org/mule/extension/ftp/FtpWriteTestCase.java

```diff
@@ -142,6 +147,26 @@ public void writeStaticContent() throws Exception
         assertThat(content, is(HELLO_WORLD));
     }
 
+    @Test
+    public void writeWithCustomEncoding() throws Exception
+    {
+        final String defaultEncoding = muleContext.getConfiguration().getDefaultEncoding();
+        assertThat(defaultEncoding, is(notNullValue()));
+
+        final String customEncoding = Charset.availableCharsets().keySet().stream()
+                .filter(encoding -> !encoding.equals(defaultEncoding))
+                .findFirst()
+                .orElse(null);
+
+        assertThat(customEncoding, is(notNullValue()));
+        final String filename = "encoding.txt";
+
+        doWrite("write", filename, HELLO_WORLD, CREATE_NEW, false, customEncoding);
+        InputStream content = (InputStream) readPath(Paths.get(testHarness.getWorkingDirectory()).resolve(filename).toString()).getPayload();
+
+        assertThat(Arrays.equals(IOUtils.toByteArray(content), HELLO_WORLD.getBytes(customEncoding)), is(true));
```

Comment:
yes, but in FTP. Although there's common ground, there're also important differences. The effort to reuse that code doesn't justify the effort. Already tried...